src/sync/tracks.nim

```diff
@@ -83,5 +83,9 @@ proc findTrackExercises(repo: TrackRepo, conf: Conf): seq[TrackExercise] =
       result.add(newTrackExercise(repoExercise))
 
 proc findTrackExercises*(conf: Conf): seq[TrackExercise] =
-  let trackRepo = newTrackRepo()
+  let trackRepo =
+    if conf.trackDir.len > 0:
+      TrackRepo(dir: conf.trackDir)
+    else:
+      newTrackRepo()
```

Comment:
I'm wondering if it would perhaps be easier to set `conf.trackDir` to the current directory if the `--track-dir` argument is specified.

Reply:
> set `conf.trackDir` to the current directory if the `--track-dir` argument is specified.

Do you mean "set the current directory to `conf.trackDir` if `--track-dir` is specified"?

We could indeed change the current directory in the program somewhere, but I think that might be harder to debug - it's more implicit.

We could use the `withDir` template, but it might be easier to just pass `conf` or `conf.trackDir` around for now. In the long-term I want to do a refactoring pass on `exercises.nim`, `tracks.nim`, and `sync.nim` that should help some of these issues. I have an old branch for that, but I never finished it.